camelot/core/naming.py

```diff
@@ -736,6 +744,56 @@ def resolve(self, name: Name) -> object:
         except (ValueError, decimal.InvalidOperation):
             raise NameNotFoundException(name[0], BindingType.named_object)
 
+class EntityNamingContext(EndpointNamingContext):
+    """
+    Represents a stateless endpoint naming context, which handles resolving instances of an Entity.
+    """
+
+    def __init__(self, entity):
+        super().__init__()
+        from camelot.core.orm import EntityBase
+        assert issubclass(entity, EntityBase)
+        self.entity = entity
+
+    @classmethod
+    def validate_atomic_name(cls, name: str) -> bool:
+        """
+        Customized atomic name validation for this entity naming context that enforces
+        the atomic names used by this context to be numeric, as they are used as primary keys
+        to query entity instances with.
+
+        :raises:
+            NamingException NamingException.Message.invalid_atomic_name_numeric when the given name is not numeric.
+        """
+        super().validate_atomic_name(name)
+        if not name.isdecimal():
+            raise NamingException(NamingException.Message.invalid_name, reason=NamingException.Message.invalid_atomic_name_numeric)
+
+    @AbstractNamingContext.check_bounded
+    def resolve(self, name: Name) -> object:
+        """
+        Resolve a name in this EntityNamingContext and return the bound object.
+        The name should be singular and its atomic form numeric, as it is used
+        as the primary key to query the corresponding instance with of the entity of this naming context.
+
+        It will throw appropriate exceptions if the resolution failed.
+
+        :param name: name under which the object should have been bound, atomic or composite, and relative to this naming context.
+
+        :return: the bound object, an instance of this EntityNamingContext's entity class.
+
+        :raises:
+            UnboundException NamingException.unbound: if this NamingContext has not been bound to a name yet.
+            NamingException NamingException.Message.invalid_name: when the name is invalid.
+            NameNotFoundException NamingException.Message.name_not_found: if no binding was found for the given name.
+        """
+        from camelot.core.orm import Session
+        name = self.get_composite_name(name)
+        instance = Session().query(self.entity).get(name[0])
```

Comment:
houd dit ook rekening met entities met een composite primary key ?

Reply:
Fixed in https://github.com/v-finance/camelot/pull/336/